src/stores/vendors/reducer.ts

```
@@ -0,0 +1,51 @@
+import { Vendor, VendorsStoreState } from '../../definitions/dokan-vendors';
+import defaultState from './defaultState';
+
+const reducer = ( state = defaultState, action: any ): VendorsStoreState => {
+    switch ( action.type ) {
+        case 'SET_VENDORS':
+            return { ...state, vendors: action.vendors };
+        case 'SET_VENDOR':
+            // Check if vendor already exists
+            const vendorExists = state.vendors.some(
+                ( vendor ) => vendor.id === action.vendor.id
+            );
```

Biome (2.1.2)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Safe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)